jax/_src/config.py

```diff
@@ -426,6 +420,17 @@ def update_thread_local_jit_state(**kw):
   tls.extra_jit_context = context._replace(**kw)
 
 
+def get_trace_state() -> Hashable:
+  # This function is used for Python-based caching (as opposed to caching in C++
+  # dispatch code). It reads the axis_env_state from the C++ thread-local state,
```

Comment:
Perhaps comment why this is the case? (i.e., there's no such thing as a global axis environment.)

Reply:
Done!